node-gui/src/main_window/main_widget/tabs/wallet.rs

```diff
@@ -0,0 +1,555 @@
+        |file_name| file_name.to_string_lossy().to_string(),
+    );
+
+    let account_items = wallet_info
+        .accounts
+        .iter()
+        .map(|(account_id, account)| AccountPickItem {
+            name: account
+                .name
+                .clone()
+                .unwrap_or_else(|| format!("Account {}", account_id.account_index())),
+            account_id: *account_id,
+        })
+        .collect::<Vec<_>>();
+
+    let selected_account = account_items
+        .iter()
+        .find(|account_item| account_item.account_id == selected_account)
+        .cloned();
+    let pick_list = pick_list(account_items, selected_account, |item| {
+        WalletMessage::SelectAccount(item.account_id)
+    });
+
+    let panel_button = |label, panel, selected_panel| {
+        let label = text(label).size(16);
+
+        let button = button(label)
+            .style(if panel == selected_panel {
+                iced::theme::Button::Primary
+            } else {
+                iced::theme::Button::Text
+            })
+            .width(Length::Fill);
+
+        button.on_press(WalletMessage::SelectPanel(panel)).padding(8)
+    };
+
+    // `next_height` is used to prevent flickering when a new block is found
+    let show_scan_progress = wallet_info.best_block.1.next_height() < node_state.best_block_height;
+    let scan_progress_widget = if show_scan_progress {
+        // TODO: Fix scan progress when the node is in the initial block download state
+        let scan_progress = wallet_scan_progress(
+            node_state.best_block_height.into_int(),
+            wallet_info.best_block.1.into_int(),
+        );
+        let scan_progress_str = format!("{:.0}%", scan_progress);
+        column![text(scan_progress_str), progress_bar(0.0..=100.0, scan_progress)]
+            .padding(10)
+            .spacing(10)
+            .align_items(Alignment::Center)
+            .width(Length::Fill)
+    } else {
+        Column::new()
+    };
+
+    column![
+        column![
+            text(file_name).size(25),
+            row![
+                pick_list.width(100),
+                button(Text::new("+"))
+                    .style(iced::theme::Button::Positive)
+                    .on_press(WalletMessage::NewAccount),
+            ]
+            .align_items(Alignment::Center)
+            .spacing(10)
+            .width(Length::Fill)
+        ]
+        .spacing(10)
+        .padding(10),
+        column![
+            panel_button("Transactions", SelectedPanel::Transactions, selected_panel),
+            panel_button("Addresses", SelectedPanel::Addresses, selected_panel),
+            panel_button("Send", SelectedPanel::Send, selected_panel),
+            panel_button("Stake", SelectedPanel::Stake, selected_panel),
+        ]
+        .height(Length::Fill),
+        scan_progress_widget,
+    ]
+    .into()
+}
+
+fn view_addresses(account: &AccountInfo) -> Element<'static, WalletMessage> {
```

Comment:
Please separate the different views into submodules so that they can be expanded.